### ISSUE TITLE
Locked User Accounts

### ISSUE THREAD

**Issue by rectifyer** [trakt/trakt-api#228]
### As of **November 9, 2020**, we’re returning an error code for locked user accounts.

-----

### What does a locked account mean?
Over the past several months, Trakt performance has been negatively impacted by a huge increase in API traffic and poorly coded apps. Some apps are sending a mass amount of duplicate data to user accounts, which causes the user's account to time out on the website and API. Rather than waste API cycles, we're flagging these accounts and will return an API error until [Trakt support](https://support.trakt.tv) fixes their account.

### What updates does my app need?
Your app will need to handle the `423` HTTP status code and should instruct the user to [contact Trakt support](https://support.trakt.tv) so we can fix their account. The [API docs](https://trakt.docs.apiary.io/#introduction/locked-user-account) have more details.

### How can I prevent duplicate data?
Please make sure your app isn't sending duplicate data. We typically see this with a user's watched history. Some apps are re-sending a user's entire watched history every day and as you can imagine, that quickly adds up to millions of duplicate plays. Other apps try and fix things by removing then re-adding all plays, but that just compounds the problem and makes things worse.

Please handle any error conditions the API sends. For example. if you get an timeout error when retrieving a user's watched history, don't simply assume all their data needs to be re-synced. Instead, the sync should be aborted and an error presented to the user with a way for them to retry later. We rely on apps to compare a user's Trakt and local data, and only sync up changes. 

### Moving Forward
We're happy to answer any questions or review code if you need help with syncing data. Just post some examples and details in this GitHub project and we'll take a look along with all the other devs.

-----

If you have any questions, please continue the discussion below.

**Comment by rudf0rd** [comment]
Furthermore, this is a relatively small amount of users so it will be quite possible you never even encounter this issue.  As I type this, it's .002% of the daily active users.

**Comment by jonathanantoine** [comment]
Hello,

A user of my app is receiving 423 status code but someone at Trakt support (Sean) told her that its account was not locked. What's happening ?

Thanks !

**Comment by UweTrottmann** [comment]
@rectifyer Does the website show if an account is locked? Until an app implements this, it could be helpful to direct users whose accounts show "symptoms" to check the website.

**Comment by jonathanantoine** [comment]
Hello @rectifyer ,

Some users don't seems to have answer to their mail related to blocked account. Is there a special email ? 

Thanks a lot !

**Comment by rectifyer** [comment]
@UweTrottmann We don't, but that is a good idea. We plan to add some more self service ways of unlocking the account or contacting us about it more easily.

@jonathanantoine same support email, we just have a huge backlog of support requests at the moment.

**Comment by jonathanantoine** [comment]
Thanks @rectifyer ! By the way do you know for sure which app created the issue in the first place ? I would like to fix it if it's from my app.

**Comment by rudf0rd** [comment]
@jonathanantoine when I clear a user's account, I tell them which app is the offending app. At this point, I recognize most by name if they've been a part of any glitches.  What's your apps name?

**Comment by jonathanantoine** [comment]
@rudf0rd my app is "TV Show Tracker - trakt client". It can set episode as watched only one by one (by batch but one by one). An user told me my app was responsible of the lock and I was wondering if it's because my app was the only displaying the message. Can you please tell me more 😀

**Comment by rudf0rd** [comment]
Ah.  Yeah I’ve seen tv show tracker come up a few times.  There’s three main reasons why we see dupes:

1. Syncing, then for some reason deleting, then resyncing.  Rinse, repeat.  When this happens, I’ll see 300k rows and 295k are deleted.
2. Server timeouts causing retries after the database statement has already been executed.  Usually from media centers, but not exclusively.  With this, I’ll see a ton of records but if I look at distinct watched at timestamps, it drops to a reasonable number (sub 50k).
3. A couple apps just went wonky and sent a bunch of bad data.  Timestamps are always set to right now so there’s no deduping possible except to go to 1 play per item.
4. Bonus possible 4th.. I don’t know this one for certain but I think it’s possible two apps fight over who is the primary data and sync over each other.  Only seen this a couple of times and it was mobile apps.

I think your app falls into #2.  I’ll see if I can find an example tomorrow.

- Sean

**Comment by jonathanantoine** [comment]
Thanks a lot Sean,

For you to knoy my app did not attempt retries until a week or so. It was a direct call on a user action (tick a box or click a button) with immediate feedback to the user.

Is it possible that when creating a new account and setting all ehe already watched shows it trigger the "locking process"? 

Thanks for your input @rudf0rd !

**Comment by rudf0rd** [comment]
@jonathanantoine 

Ok I have one I'm looking at right now.  Here's some example data:

* User has 386k rows created by tv show tracker (looks like they only use your app)
* Of the 386k, there's 16k real plays that are duplicated a bunch
* [Here's a quick excerpt of their sync history](http://pics.rudford.com/Screen-Shot-2021-01-22-09-07-40.12.png) by when each play record was created (y = year, m = month, d = day),  You can see there's a few oddball spikes which could be explained by marking a couple series entirely watched (simpsons, etc), but the real damage seems to have started on the 11th.  The account was then locked on the 17th.

During this time period, we really didn't have any large server issues like we were seeing back in October/November, so I don't think is timeout/retry from a 5xx server response.  Did something change in your app around this time?

**Comment by jonathanantoine** [comment]
@rudf0rd not real change... that's really strange. Can this be a webhook ? Do you know if it's always the same items being set as played ?

**Comment by rudf0rd** [comment]
@jonathanantoine when another case comes up, i'll check to see if there's any pattern to what's being repeated, but it seems like it's fairly widespread across the library

**Comment by jonathanantoine** [comment]
@rudf0rd Can you tell me if you still have this behavior for my app ? Now the only time it send again a "set as seen" command is if the APi does not answer by a 200 status code.

**Comment by rudf0rd** [comment]
@jonathanantoine yes, most definitely.  I sent you an email on Monday (i think) regarding the incredibly high duplicate rates (in the millions).

**Comment by jonathanantoine** [comment]
@rudf0rd I didn't received it :( can you send it at apps [at] jonathanantoine.com ?
I would to be able to chat with you via a more direct method (discord, etc.) if that don't bother you. I don't like the idea of spamming your API and I would love to fix it ASAP.

**Comment by Gibby** [comment]
Was there a special email made yet by chance for requesting help with a locked account? I contacted support last week using the link in OP.

**Comment by KathleenMc73** [comment]
I am getting a message saying my account is locked. Can I get it unlocked please.
Also, having installed the TV tracker app onto my new phone I am getting a message saying sync failure so none of my inputted data is appearing.

**Comment by vlevas** [comment]
Hello i also have problem and send a ticket but nothing yet :/
@rudf0rd is possible to help me? the username is: valantislevas i already send a ticket

**Comment by trevevs** [comment]
hey, this happened to me too. cant see any way to raise a trakt support ticket- they removed it?  how do I fix?  thanks

**Comment by rectifyer** [comment]
You can use the [support portal](https://support.trakt.tv) or email support@trakt.tv directly.
